pkg/lib/aws/errors.go

```diff
@@ -175,3 +180,38 @@ func ErrorDashboardHeightOutOfRange(height int) error {
 		Message: fmt.Sprintf("dashboard height %d out of range; height must be between %d and %d", height, _dashboardMinHeightUnits, _dashboardMaxHeightUnits),
 	})
 }
+
+func ErrorNATGatewayLimitExceeded(currentLimit, additionalQuotaRequired int, availabilityZones []string, region string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrNATGatewayLimitExceeded,
+		Message: fmt.Sprintf("NAT gateway limit of %d exceeded in availability zones (%s) of region %s; increase the quota for NAT gateways by at least %d or remove some of the existing ones", currentLimit, s.StrsAnd(availabilityZones), region, additionalQuotaRequired),
+	})
+}
+
+func ErrorNATGatewayLimitExceededInAllAZs(currentLimit, additionalQuotaRequired int, region string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrNATGatewayLimitExceededInAllAZs,
+		Message: fmt.Sprintf("NAT gateway limit of %d exceeded in all availability zones of region %s; increase the quota for NAT gateways by at least %d or remove some of the existing ones", currentLimit, region, additionalQuotaRequired),
```

Comment:
Is it possible that this error message could be misleading, e.g. if your cluster is in 3 AZs which are all full, but there are two more AZs in your region (which are not used by your cluster) but have availability? If so, would it make sense to remove this error and reuse the one above in both cases?

Reply:
Yes, I see. That makes sense.